lib/carbonite.ex

```diff
@@ -11,8 +11,185 @@ defmodule Carbonite do
+  """
+  @doc since: "0.4.0"
+  @spec override_mode(repo()) :: :ok
+  @spec override_mode(repo(), [prefix_option()]) :: :ok
+  def override_mode(repo, opts \\ []) do
+    carbonite_prefix = Keyword.get(opts, :carbonite_prefix, default_prefix())
+
+    from(t in Trigger,
+      update: [set: [override_transaction_id: fragment("pg_current_xact_id()")]]
+    )
+    |> repo.update_all([], prefix: carbonite_prefix)
+
+    :ok
+  end
+
+  @type process_option ::
+          Carbonite.Query.outbox_queue_option()
+          | {:batch_filter, (Ecto.Query.t() -> Ecto.Query.t())}
+  @type process_func :: (Transaction.t(), Outbox.memo() -> Outbox.memo())
+
+  @doc """
+  Processes an outbox queue.
+
+  See `Carbonite.Query.outbox_queue/2` for query options.
+
+  ## Parameters
+
+  * `repo` - the Ecto repository
+  * `outbox_name` - name of the outbox to process
+  * `opts` - optional keyword list
+
+  ## Options
+
+  * `min_age` - the minimum age of a record, defaults to 300 seconds (set nil to disable)
+  * `batch_size` - limits the query in size, defaults to 100 (set nil to disable)
+  * `batch_filter` - function for refining the batch query, defaults to nil
+  * `carbonite_prefix` - defines the audit trail's schema, defaults to `"carbonite_default"`
+  """
+  @doc since: "0.4.0"
+  @spec process(repo(), Outbox.name(), process_func()) :: :ok
+  @spec process(repo(), Outbox.name(), [process_option()], process_func()) :: :ok
+  def process(repo, outbox_name, opts \\ [], process_func) do
+    carbonite_prefix = Keyword.get(opts, :carbonite_prefix, default_prefix())
+    batch_filter = Keyword.get(opts, :batch_filter) || (& &1)
+
+    outbox =
+      outbox_name
+      |> Carbonite.Query.outbox(opts)
+      |> repo.one!()
+
+    attrs =
+      outbox
+      |> Carbonite.Query.outbox_queue(opts)
+      |> batch_filter.()
+      |> repo.all()
+      |> process_batch(outbox, process_func)
+
+    outbox
+    |> Outbox.changeset(attrs)
+    |> repo.update!(prefix: carbonite_prefix)
+
+    :ok
+  end
+
+  defp process_batch(batch, outbox, process_func) do
+    initial = Map.take(outbox, [:last_transaction_id, :memo])
+
+    Enum.reduce(batch, initial, fn transaction, acc ->
+      %{
+        last_transaction_id: transaction.id,
+        memo: process_func.(transaction, acc.memo)
+      }
+    end)
+  end
+
+  @type purge_option :: Carbonite.Query.outbox_done_option()
+
+  @doc """
+  Deletes transactions that have been fully processed.
+
+  See `Carbonite.Query.outbox_done/1` for query options.
+
+  ## Parameters
+
+  * `repo` - the Ecto repository
+  * `opts` - optional keyword list
+
+  ## Options
+
+  * `min_age` - the minimum age of a record, defaults to 300 seconds (set nil to disable)
+  * `carbonite_prefix` - defines the audit trail's schema, defaults to `"carbonite_default"`
+  """
+  @doc since: "0.4.0"
+  @spec purge(repo()) :: :ok
+  @spec purge(repo(), [purge_option()]) :: :ok
+  def purge(repo, opts \\ []) do
```

Comment:
I am confused by this one. I was expecting it to accept an outbox as a parameter to work on 🤔 (so like purge everything that went through this outbox)

Reply:
Imho this should be `purge(repo, outbox_name, opts)`

Reply:
But if I had two outboxes and one is behind, we would delete transactions it hasn't seen yet?!